pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -0,0 +1,316 @@
+import asyncio
+import glob
+import json
+import os
+
+import click
+import yaml
+
+from pyartcd.cli import cli, pass_runtime, click_coroutine
+from pyartcd.runtime import Runtime
+from pyartcd import exectools
+from pyartcd.util import branch_arches
+
+GO_ARCH_SUFFIXES = {
+    'x86_64': '',
+    's390x': '-s390x',
+    'ppc64le': '-ppc64le',
+    'aarch64': '-arm64'
+}
+GEN_PAYLOAD_ARTIFACTS_OUT_DIR = 'gen-payload-artifacts'
+
+
+class BuildSyncPipeline:
+    def __init__(self, runtime: Runtime, version: str, assembly: str, publish: bool, data_path: str,
+                 emergency_ignore_issues: bool, retrigger_current_nightly: bool, doozer_data_gitref: str,
+                 debug: bool, images: str, exclude_arches: str, skip_multiarch_payload: bool):
+        self.runtime = runtime
+        self.version: str = version
+        self.assembly: str = assembly
+        self.publish: bool = publish
+        self.data_path: str = data_path
+        self.emergency_ignore_issues: bool = emergency_ignore_issues
+        self.retrigger_current_nightly: bool = retrigger_current_nightly
+        self.doozer_data_gitref: str = doozer_data_gitref
+        self.debug: bool = debug
+        self.images: str = images
```

Comment:
Types can be inferred here. No need to be explicitly defined.

Reply:
Updated in 3d826f0613434e304f12d57866a0cd97b2d7c78a